cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1049 @@
+
+    threshold:
+        Label quality scores above the threshold are not considered as issues and their indices are omited from the return
+
+    Returns
+    ---------
+    issue_indices:
+        Array of issue indices sorted from most to least severe who's label quality scores fall below the threshold if one is provided.
+    """
+
+    if threshold > 1.0:
+        raise ValueError(
+            f"""
+            Threshold is a cutoff of label_quality_scores and therefore should be <= 1.
+            """
+        )
+
+    issue_indices = np.argwhere(label_quality_scores <= threshold).flatten()
+    issue_vals = label_quality_scores[issue_indices]
+    sorted_idx = issue_vals.argsort()
+    return issue_indices[sorted_idx]
+
+
+def _compute_label_quality_scores(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    *,
+    method: str = "subtype",
+    threshold: Optional[float] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Internal function to prune extra bounding boxes and compute label quality scores based on passed in method."""
+
+    pred_probs_prepruned = False
+    min_pred_prob = _get_min_pred_prob(predictions)
+
+    if threshold is not None:
+        predictions = _prune_by_threshold(
+            predictions=predictions, threshold=threshold, verbose=verbose
+        )
+        if np.abs(min_pred_prob - threshold) < 0.001 and threshold > 0:
+            pred_probs_prepruned = True  # the provided threshold is the threshold used for pre_pruning the pred_probs during model prediction.
+    else:
+        threshold = min_pred_prob  # assume model was not pre_pruned if no threshold was provided
+
+    if method == "subtype":
+        alpha = 0.91
+        low_probability_threshold = 0.001
+        high_probability_threshold = 0.5
+        temperature = 0.1
+        scores = _get_subtype_label_quality_scores(
+            labels,
+            predictions,
+            alpha=alpha,
+            low_probability_threshold=low_probability_threshold,
+            high_probability_threshold=high_probability_threshold,
+            temperature=temperature,
+        )
+
+    return scores
+
+
+def _get_min_pred_prob(
+    predictions: List[np.ndarray],
+) -> float:
+    """Returns min pred_prob out of all predictions."""
+    pred_probs = [1.0]  # avoid calling np.min on empty array.
+    for prediction in predictions:
+        for class_prediction in prediction:
+            pred_probs.extend(list(class_prediction[:, -1]))
+
+    min_pred_prob = np.min(pred_probs)
+    return min_pred_prob
+
+
+def _prune_by_threshold(
+    predictions: List[np.ndarray], threshold: float, verbose: bool = True
+) -> List[np.ndarray]:
+    """Removes predicted bounding boxes from predictions who's pred_prob is below the cuttoff threshold."""
+
+    max_allowed_box_prune = 0.97  # This is max allowed percent of prune for boxes below threshold before a warning is thrown.
```

Comment:
move to constants, clarify in comment that its for finding label issues in object detection, and name in all caps